backend/routes/institution/institution_router.py

```
@@ -0,0 +1,421 @@
+    except Exception as e:
+        logger.error(f"Error running simulation: {e}")
+        return ErrorResponseModel(
+            status="error", message=f"Failed to run simulation: {str(e)}"
+        )
+
+
+@institution_router.post("/get-all-league-results", response_model=ResponseModel)
+@verify_admin_or_institution
+async def get_league_results_endpoint(
+    league: LeagueName,  # Change from LeagueSignUp to LeagueName
+    current_user: dict = Depends(get_current_user),
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)